watcher/lib/influxdb/rollups.go

```diff
@@ -126,29 +126,56 @@ func buildFunction(f Function) (string, error) {
 		alias = f.Field
 	}
 	if strings.HasPrefix(f.Function, constants.FunctionPercentile) {
-		value, err := parsePercentileValue(f.Function)
+		value, err := parseValueComposedFunc(f.Function)
 		if err != nil {
 			return "", trace.Wrap(err)
 		}
 		return fmt.Sprintf("%v(%v, %v) as %v", constants.FunctionPercentile, f.Field, value, alias), nil
 	}
+	if strings.HasPrefix(f.Function, constants.FunctionBottom) {
```

Comment:
This does not look composable. If the function and the value are always encoded as `func_value`, then the parsing can start earlier:
```go
// buildFunction returns a function string based on the provided function configuration
func buildFunction(f Function) (string, error) {
	alias := f.Alias
	if alias == "" {
		alias = f.Field
	}

	funcAndValue := strings.Split(f.Function, "_")
	if len(funcAndValue) == 2 && isFuncWithParams(funcAndValue[0]) {
		funcName := funcAndValue[0]
		param := funcAndValue[1]
		err := validateParam(funcName, param)
		if err != nil {
			return "", trace.Wrap(err)
		}
		return fmt.Sprintf("%v(%v, %v) as %v", funcName, f.Field, param, alias), nil
	}

	return fmt.Sprintf("%v(%v) as %v", f.Function, f.Field, alias), nil
}

var funcsWithParams = []string{
	constants.FunctionPercentile,
	constants.FunctionBottom,
	constants.FunctionTop,
	constants.FunctionSample,
}

// validateParam checks the function parameter for validity.
func validateParam(funcName, param string) error {
	value, err := strconv.Atoi(param)
	if err != nil {
		return trace.Wrap(err)
	}

	switch funcName {
	case constants.FunctionPercentile:
		if value < 0 || value > 100 {
			return trace.BadParameter(
				"percentile value must be between 0 and 100 (inclusive)")
		}
	case constants.FunctionTop, constants.FunctionBottom, constants.FunctionSample:
		if value < 0 {
			return trace.BadParameter(
				"Top, Bottom and Sample value must be greater or equal to 0")
		}
	}

	return nil
}
```